hooks/gen_docs.py

```diff
@@ -1,10 +1,308 @@
 """Generates the whole 'generatable' KPOps documentation"""
+import logging
 import subprocess
+import sys
+from pathlib import Path
+from typing import NamedTuple, cast
+
+import yaml
 
 from hooks import PATH_ROOT
+from kpops.cli.registry import _find_classes
+from kpops.components import KafkaConnector, PipelineComponent
+from kpops.components.base_components.base_defaults_component import deduplicate
+from kpops.utils.colorify import redify, yellowify
+from kpops.utils.yaml_loading import load_yaml_file
 
 PATH_KPOPS_MAIN = PATH_ROOT / "kpops/cli/main.py"
 PATH_CLI_COMMANDS_DOC = PATH_ROOT / "docs/docs/user/references/cli-commands.md"
+PATH_DOCS_RESOURCES = PATH_ROOT / "docs/docs/resources"
+PATH_DOCS_COMPONENTS = PATH_DOCS_RESOURCES / "pipeline-components"
+PATH_DOCS_COMPONENTS_DEPENDENCIES = (
+    PATH_DOCS_COMPONENTS / "dependencies/pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_COMPONENTS_DEPENDENCIES_DEFAULTS = (
+    PATH_DOCS_COMPONENTS / "dependencies/defaults_pipeline_component_dependencies.yaml"
+)
+PATH_DOCS_KPOPS_STRUCTURE = PATH_DOCS_COMPONENTS / "dependencies/kpops_structure.yaml"
+KPOPS_COMPONENTS = tuple(_find_classes("kpops.components", PipelineComponent))
+KPOPS_COMPONENTS_INHERITANCE_REF = {
+    component.get_component_type(): cast(
+        type[PipelineComponent], component.__base__
+    ).get_component_type()
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_SECTIONS = {
+    component.get_component_type(): [
+        field_name
+        for field_name, model in component.__fields__.items()
+        if not model.field_info.exclude
+    ]
+    for component in KPOPS_COMPONENTS
+}
+KPOPS_COMPONENTS_SECTIONS_ORDER = deduplicate(
+    [field for fields in KPOPS_COMPONENTS_SECTIONS.values() for field in fields]
```

Comment:
could you extract a `flatten` function from this list comprehension? it's a little hard to grasp

Reply:
it could live in the same location as the `deduplicate` function and ideally that's not the current location `base_defaults_component.py` 😄

Reply:
Sure

Reply:
I ended up removing it instead. The sort no longer made sense without any sets and with your reordering PR